core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -63,20 +63,25 @@ private[spark] class CoarseMesosSchedulerBackend(
   // Maximum number of cores to acquire (TODO: we'll need more flexible controls here)
   val maxCores = conf.get("spark.cores.max",  Int.MaxValue.toString).toInt
 
+  val maxExecutorsPerSlave = conf.getInt("spark.mesos.coarse.executors.max", 1)
+  val maxCpusPerExecutor = conf.getInt("spark.mesos.coarse.cores.max", Int.MaxValue)
```

Comment:
It's quite hard to differentiate between spark.cores.max since spark.cores itself is already very vague, where it's a configuration to set the total number of cores a Spark app can schedule.
spark.mesos.coarse.cores.max is the Maximum cores a coarse grained Spark executor can take up to, and the scheduler will schedule any cores between 1 to spark.mesos.coarse.cores.max. 

So calling it coresPerExecutor doesn't seem right as it's not a hard value that the scheduler tries to schedule.

How about spark.mesos.coarse.coresPerExecutor.max?